src/olympia/reviewers/utils.py

```diff
@@ -842,22 +844,27 @@ def log_action(
             'reviewtype': self.review_type.split('_')[1],
             **(extra_details or {}),
         }
-        if file is None and self.file:
-            file = self.file
-        if file is not None:
-            details['files'] = [file.id]
         if version is None and self.version:
             version = self.version
+
+        if file is not None:
+            details['files'] = [file.id]
+        elif self.file:
+            details['files'] = [self.file.id]
```

Comment:
nit, and probably out of scope for this patch, but I can't see why we'd need to pass `file` to this function these days - it will always be `version.file`

Reply:
https://github.com/mozilla/addons-server/issues/20442